databaseutil/databaseutil.go

```diff
@@ -94,6 +94,59 @@ func GetPasswordForUserWithEmailAddress(emailAddress string) ([]byte, error) {
 	return password, nil
 }
 
+func StoreNewNote(authorId int64, content string, creationTime time.Time) (int64, error) {
+	sqlQuery := `
+		INSERT INTO note (author_id, content, creation_time)
+		VALUES ($1, $2, $3)
+		RETURNING id`
+
+	rows, err := db.Query(sqlQuery, authorId, content, creationTime)
+	if err != nil {
+		return -1, convertPostgresError(err)
```

Comment:
I believe our/go convention is to return 0, not -1

Reply:
You're right, I'll change it.  I just makes me feel safer returning an invalid value on errors.

Reply:
how are you defining invalid value?

Reply:
it depends on the application and not everything has an invalid value (which is the crux of the issue). But for sql id a negative value is obviously invalid. But I changed it to zero, because that's convention.